### DISCUSSION TITLE
Issue : File validation fails only in functional tests

### DISCUSSION THREAD

**Discussion by RoyalVICTOR-pro** [adonisjs/core#4276] [Help]
Hello, I'm developing an API with AdonisJS 5. I have a route for creating a profession. This profession comes with 2 files. I validate the data using the following validator:

```typescript
  public schema = schema.create({
    name_fr: schema.string({}, [
      rules.unique({ table: 'professions', column: 'name_fr' }), 
    ]),
    picto_file: schema.file.optional({
      size: '2mb', 
      extnames: ['jpg', 'png', 'jpeg'],
    }),
    image_file: schema.file.optional({
      size: '2mb',
      extnames: ['jpg', 'png', 'jpeg'],
    }),
    is_enabled: schema.boolean.optional(),
  })
```

When I test the POST request with the files via Postman, there are no issues.

However, when I try to do the same thing from a Japa integration test, it no longer works. 
The validation fails, and I get a 422 status code.

While trying to add rules one by one, I realized that it's the size validation that is failing. Both of my files are just a few bytes, and they are the same files I use via Postman and in the test, so the problem doesn't come from there.

Here's my test (which works fine if the file size validation is not included):"

```typescript
test('Create a first profession with valid data with admin role', async ({ client }) => {
    const response = await client
      .post('/professions')
      .bearerToken(fakeUser.adminToken)
      .file('picto_file', picto1Path)
      .file('image_file', image1Path)
      .fields({ name_fr: 'Profession Test 1', is_enabled: true })

    response.assertStatus(201)
    response.assertBodyContains({
      name_fr: 'Profession Test 1',
      picto_file: './professions/pictos/profession-test-1.jpg',
    })
    firstProfessionId = response.body().id
  })
```

Does this ring a bell for anyone? If anyone has a solution, I'm all ears!

Thank you very much!

**Comment by RoyalVICTOR-pro** [comment]
Is there a cache or something similar? Because I went to have lunch after creating the ticket, and when I reran the tests with the 'size' line added back, it worked right away. I didn't touch anything else.

**Reply by thetutlage** [reply]
Nope. There is no cache anywhere
